vncviewer/Viewport.cxx

```diff
@@ -606,6 +606,14 @@ int Viewport::handle(int event)
     if (Fl::event_button3())
       buttonMask |= 4;
 
+    //Fl::event_button is only good for FL_PUSH and FL_RELEASE
+    if(event == FL_PUSH) {
+      if (Fl::event_button() == 8) //Mouse Back
+        buttonMask |= 1024;
+      if (Fl::event_button() == 9) //Mouse Forward
+        buttonMask |= 512;
+    }
```

Comment:
This does not work for me here with Fedora 38. I get button 10 and 11 in the session, instead of the expected 8 and 9.

Reply:
I feared this would be the case. What would be the solution here, read a config file for the button mapping?

Reply:
That should not be necessary. Applications such as Chrome and Firefox do not have a config file for this, so it shouldn't be necessary for us.

Where did you get the values 1024 and 512?

Reply:
the way the buttons are mapped from the input is a bit shift in vncPointerButtonAction. Back is the 10th button in the map so 1<<10.

Reply:
Do you mean it's the one labeled `BTN_LABEL_PROP_BTN_BACK`? Unfortunately, that label is wrong. You'll need to compare with the events you get locally, since that is what the applications expect. Things such as `xev` should look identical locally and in the VNC session.

And no, we don't want to fix the labels. We need to be bug-for-bug compatible with a local system to avoid breaking applications.